src/history/HistoryManagerImpl.cpp

```diff
@@ -220,8 +220,14 @@ HistoryManagerImpl::queueCurrentHistory()
 {
     ZoneScoped;
     auto ledger = mApp.getLedgerManager().getLastClosedLedgerNum();
-    HistoryArchiveState has(ledger, mApp.getBucketManager().getBucketList(),
-                            mApp.getConfig().NETWORK_PASSPHRASE);
+
+    BucketList bl;
+    if (mApp.getConfig().MODE_ENABLES_BUCKETLIST)
+    {
+        bl = mApp.getBucketManager().getBucketList();
+    }
+
+    HistoryArchiveState has(ledger, bl, mApp.getConfig().NETWORK_PASSPHRASE);
```

Comment:
Then `HistoryArchiveState` will be constructed by a default initialized `BucketList`. This is ok, since the default initialize just creates a complete BucketList, where each Bucket is empty. This is a valid state so we shouldn't hit any bad asserts.